src/menu/views/system_info.c

```diff
@@ -75,11 +77,12 @@ static void draw (menu_t *menu, surface_t *d) {
 }
 
 
-void view_system_info_init (menu_t *menu) {    
-    controllers = get_controllers_present();
-
+void view_system_info_init (menu_t *menu) {
     for (int i = 0; i < 4; i++) {
-        accessory[i] = identify_accessory(i);
+        joypad[i] = joypad_is_connected(i);
```

Comment:
If you want to make sure the port actually has a controller connected (as opposed to something like a VRU), you may want to check the Joypad style instead of using `joypad_is_connected`:

```c
joypad[i] = joypad_get_style(i) != JOYPAD_STYLE_NONE;
```